### PR TITLE
Add moshi

### PR DESCRIPTION
**Pull request by UserNameMax**
Replace gson on moshi in leader

### CODE REVIEW

File: tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/LeaderApi.kt [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 
+import band.effective.office.tv.network.leader.models.Response
 import retrofit2.Call
 import retrofit2.http.GET
 import retrofit2.http.Query
 
 interface LeaderApi {
     @GET("api/v4/events/search")
-    fun searchEvents(@Query("paginationSize") count: Int, @Query("placeIds[]") placeId: Int): Call<SearchResponse>
+    fun searchEvents(@Query("paginationSize") count: Int, @Query("placeIds[]") placeId: Int): Call<Response>
```
Comment (Radch-enko) [inline tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/LeaderApi.kt:10]: `Call<SearchEventsResponse>` will be best name

File: tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/LeaderApi.kt [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
     @GET("api/v4/events/search")
-    fun searchEventsByCity(@Query("paginationSize") count: Int, @Query("cityId") cityId: Int): Call<SearchResponse>
+    fun searchEventsByCity(@Query("paginationSize") count: Int, @Query("cityId") cityId: Int): Call<Response>
```
Comment (Radch-enko) [inline tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/LeaderApi.kt:12]: There are two functions for the same endpoint is not good I think. 

Why not just add all the parameters to one function above?

File: tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/models/Response.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package band.effective.office.tv.network.leader.models
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class Response(
+    //@Json(name = "`data`")
+    val `data`: DataX
```
Comment (Radch-enko) [inline tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/models/Response.kt:9]: Why data is wrapped to `` ?

File: tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/models/ThumbX.kt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package band.effective.office.tv.network.leader.models
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class ThumbX(
+    //@Json(name = "360")
+    val `360`: String,
```
Comment (Radch-enko) [inline tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/models/ThumbX.kt:9]: Bad name. 
Let's define like this: `size360` or `thumbnail360` or `thumb360` , but not only number in name

Comment (Radch-enko) [reply tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/models/ThumbX.kt:9]: Let's avoid names that must be wrapped to ``

File: tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/LeaderApi.kt (hunk 2 above)
Comment (Radch-enko) [reply tv-app/effecticeTV/app/src/main/java/band/effective/office/tv/network/leader/LeaderApi.kt:12]: Yes, but this logic can be moved from api service like this:

For example: 
1. Define these methods on Repository level
2. Each function call only one `searchEvents` from `LeaderApi` but provide different parameters set

**Comment by UserNameMax** [comment]
add changes

**Review by Radch-enko — APPROVED** [review]

**Pull request merged**